hawkular-inventory-api/src/main/java/org/hawkular/inventory/base/BaseTenants.java

```diff
@@ -46,7 +46,7 @@ private BaseTenants() {
 
     }
 
-    public static class ReadWrite<BE> extends Mutator<BE, Tenant, Tenant.Blueprint, Tenant.Update>
+    public static class ReadWrite<BE> extends Mutator<BE, Tenant, Tenant.Blueprint, Tenant.Update, String>
```

Comment:
> That's the actual java.lang.String

oh, ok it's confusing. So if the type parameter is has the same name as imported class or a class from `java.lang.` (which are implicitly imported), it's not type parameter but a concrete class..

> Resources.Single#data(DataEntity.Role) method. This would IMHO be maybe better than the current approach of a method per data role. 

yeah, Resources.Single#data(DataEntity.Role) sounds more extensible